oidc-configure.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# Add a new realm
+cd /opt/keycloak-4.5.0.Final/bin/
+./kcadm.sh config credentials --server http://localhost:8080/auth --realm master --user admin  --password KEYCLOAKPASS
+./kcadm.sh create realms -s realm=ondemand -s enabled=true -s loginWithEmailAllowed=false -s rememberMe=true
+
+# Configure LDAP
+REALMID=$(./kcadm.sh get realms/ondemand --fields id | egrep -v '{|}' | sed 's/.*id".*:\s*"//g; s/"//g')
+./kcadm.sh create components -r ondemand -s name=ldap -s providerId=ldap -s providerType=org.keycloak.storage.UserStorageProvider -s parentId=$REALMID  -s 'config.importUsers=["false"]'   -s 'config.priority=["1"]' -s 'config.fullSyncPeriod=["-1"]' -s 'config.changedSyncPeriod=["-1"]' -s 'config.cachePolicy=["DEFAULT"]' -s config.evictionDay=[] -s config.evictionHour=[] -s config.evictionMinute=[] -s config.maxLifespan=[] -s 'config.batchSizeForSync=["1000"]'  -s 'config.editMode=["READ_ONLY"]'  -s 'config.syncRegistrations=["false"]'  -s 'config.vendor=["other"]'  -s 'config.usernameLDAPAttribute=["uid"]' -s 'config.rdnLDAPAttribute=["uid"]' -s 'config.uuidLDAPAttribute=["entryUUID"]'  -s 'config.userObjectClasses=["posixAccount"]' -s 'config.connectionUrl=["ldaps://openldap1.infra.osc.edu:636 ldaps://openldap2.infra.osc.edu:636"]' -s 'config.usersDn=["ou=People,ou=hpc,o=osc"]'  -s 'config.authType=["simple"]'  -s 'config.bindDn=["uid=admin,ou=system"]' -s 'config.bindCredential=["secret"]'  -s 'config.useTruststoreSpi=["never"]'  -s 'config.connectionPooling=["true"]' -s 'config.pagination=["true"]'
```

Comment:
I'd recommend passing `--id`.  When I use JSON to add resources with Puppet I set ID to resource name rather than having it be some random UUID.

Reply:
correction `-s id=ondemand-ldap` , can't conflict with realm ID I think.